library/cc/stream_callbacks.cc

```diff
@@ -24,9 +24,9 @@ void* c_on_headers(envoy_headers headers, bool end_stream, envoy_stream_intel in
     }
     auto on_headers = stream_callbacks->on_headers.value();
     on_headers(builder.build(), end_stream, intel);
-  } else {
-    release_envoy_headers(headers);
   }
+
+  release_envoy_headers(headers);
```

Comment:
I think that the convention in here is that `on_headers` should do the release hence we used to have an `else` statement in here

Reply:
on_headers doesn't receive the envoy_header struct, it gets the ResponseHeader type which doesn't allow it to free it. This means that it's on c_on_headers to free it regardless of the path it takes